posts/physical-backup-restore.md

```diff
@@ -0,0 +1,457 @@
+---
+title: "Physical Backup & Restoration"
+description: "Physical Backup & Restoration"
+date: "2025-04-09"
+is_redirect: false
+redirect_url:
+---
+
+
+In Frappe Cloud, every **Site Update** triggers an automatic backup - ensuring your data stays safe in case of failures and we can rollback your site to older version as soon as possible.
+
+As of now, we use `mysqldump` to take logical backup your site's database.
+
+```
+mysqldump \
+  --single-transaction --quick \
+  --lock-tables=false \
+  -u root -psecret \
+  _c678923hj223 \
+  > backup.sql
+```
+
+This works fine !
+
+### Issues
+
+1. **High Disk I/O :** When backing up your large database, MariaDB needs to read all data from disk and convert each table/row into SQL queries. As your database grows:
+	- The database need to perform more work
+	- Required more disk i/o
+	- Impact overall server performance
+
+2. **Backup Time :** In `mysqldump` based backup process, the duration depends completely on database size.
+
+	Each operation requires :
+	- Reading data from disk
+	- Converting each table definition and rows to SQL queries
+	- Write those data to \*.sql file.
+
+	So, the backup duration grows proportionally with database size.
+
+	Take a glimpse of that -
+	![](https://frappe.io/files/logical-backup.png)
+
+	If you have some site with more than 10GB of database size, you need to wait ~10 minutes before even starting with actual update.
+
+
+	> BTW, Frappe Cloud gives you an option to **Skip Backup** as well. 
+	> 
+	> But that's like playing with 🔥 in a production site. If something goes wrong, you need to fix it on production yourself. Resulted in more downtime.
+
+    <p align="center">
+        <img style="max-width: 300px" src="/files/meme435384.png" />
+    </p>
+
+
+### Simple Solution - Take Disk Snapshot
+
+Just take disk snapshot of database server because AWS EBS Snapshot, ZFS Snapshot is just a few seconds operation.
```

Comment:
In this section, I'd suggest grouping these single sentences into 2-3 coherent paragraphs. It's easier to read than separate lines. 

Essentially:
- Setting up an obvious solution
- Why that obvious solution doesn't work for us